tests/devices/default_qubit/test_default_qubit_native_mcm.py

```diff
@@ -614,3 +601,45 @@ def circuit():
 
     with pytest.raises(ValueError, match="Returning qml.sample is not supported when"):
         _ = circuit()
+
+
+@pytest.mark.jax
+@pytest.mark.parametrize("shots", [100, [100, 101]])
+@pytest.mark.parametrize("postselect", [None, 0, 1])
+@pytest.mark.parametrize("reset", [False, True])
+def test_sample_with_prng_key(shots, postselect, reset):
+    """Test that setting a PRNGKey gives the expected behaviour"""
```

Comment:
I'm a little confused by this test. Can you expand the docstring a bit and add a few inline comments so its easier to follow what the expected behaviour is and how we're verifying it?